test/Microsoft.AspNetCore.Mvc.IntegrationTests/HeaderModelBinderIntegrationTest.cs

```diff
@@ -301,8 +303,194 @@ private class ListContainer2
-            Assert.Equal(value, entry.Value.AttemptedValue);
+            Assert.Equal(expectedAttemptedValue, entry.Value.AttemptedValue);
             Assert.Equal(expectedRawValue, entry.Value.RawValue);
         }
+
+        [Fact]
+        public async Task BindPropertyFromHeader_WithPrefix_GetsBound_ForSimpleTypes()
+        {
+            // Arrange
+            var parameterBinder = ModelBindingTestHelper.GetParameterBinder();
+            var parameter = new ParameterDescriptor()
+            {
+                Name = "Parameter1",
+                BindingInfo = new BindingInfo()
+                {
+                    BinderModelName = "prefix",
+                },
+                ParameterType = typeof(Product)
+            };
+
+            var testContext = ModelBindingTestHelper.GetTestContext(
+                request =>
+                {
+                    request.Headers.Add("NoCommaString", "someValue");
+                    request.Headers.Add("OneCommaSeparatedString", "one, two, three");
+                    request.Headers.Add("IntProperty", "10");
+                    request.Headers.Add("NullableIntProperty", "300");
+                    request.Headers.Add("ArrayOfString", "first, second");
+                    request.Headers.Add("EnumerableOfDouble", "10.51, 45.44");
+                    request.Headers.Add("ListOfEnum", "Sedan, Coupe");
+                    request.Headers.Add("ListOfOrderWithTypeConverter", "10");
+                });
+            var modelState = testContext.ModelState;
+
+            // Act
+            var modelBindingResult = await parameterBinder.BindModelAsync(parameter, testContext);
+
+            // Assert
+
+            // ModelBindingResult
+            Assert.True(modelBindingResult.IsModelSet);
+
+            // Model
+            var product = Assert.IsType<Product>(modelBindingResult.Model);
+            Assert.NotNull(product);
+            Assert.NotNull(product.Manufacturer);
+            Assert.Equal("someValue", product.Manufacturer.NoCommaString);
+            Assert.Equal("one, two, three", product.Manufacturer.OneCommaSeparatedString);
+            Assert.Equal(10, product.Manufacturer.IntProperty);
+            Assert.Equal(300, product.Manufacturer.NullableIntProperty);
+            Assert.Null(product.Manufacturer.NullableLongProperty);
+            Assert.Equal(new[] { "first", "second" }, product.Manufacturer.ArrayOfString);
+            Assert.Equal(new double[] { 10.51, 45.44 }, product.Manufacturer.EnumerableOfDouble);
+            Assert.Equal(new CarType[] { CarType.Sedan, CarType.Coupe }, product.Manufacturer.ListOfEnum);
+            var orderWithTypeConverter = Assert.Single(product.Manufacturer.ListOfOrderWithTypeConverter);
+            Assert.Equal(10, orderWithTypeConverter.Id);
+
+            // ModelState
+            Assert.True(modelState.IsValid);
+            var entry = modelState["prefix.Manufacturer.NoCommaString"];
```

Comment:
Use helper methods and `Assert.Collection(...)` to shorten this and make it easier to read.